common/lcs/src/de.rs

```diff
@@ -132,8 +132,27 @@ impl<'de> Deserializer<'de> {
         Ok(u128::from_le_bytes(le_bytes))
     }
 
+    fn parse_u32_as_uleb128(&mut self) -> Result<u32> {
+        let mut value: u32 = 0;
+        let mut shift: u8 = 0;
+        loop {
+            let byte = self.next()?;
+            let num = byte & 0x7f;
+            value |= u32::from(num) << shift;
+            if num == byte {
+                // Highest bit of `byte` was 0.
+                return Ok(value);
+            }
+            shift += 7;
+            if shift > 28 {
+                break;
+            }
+        }
```

Comment:
```suggestion
    for shift in &[0, 7, 14, 21] {
        let byte = self.next()?;
        let num = byte & 0x7f;
        value |= u32::from(num) << shift;
        if num == byte {
            return Ok(value);
        }
    }
}
```

Reply:
We can definitely replace `loop {` with `while {`. I've also added a suggestion for using `for` to further make the loop concise.

Reply:
ok. Enumerating the values is a bit too much for me but why not a while loop indeed.

Reply:
In the hand I went for `for shift in (0..32).step_by(7)` :)

Reply:
@vgao1996 @davidiw By the way, now that I think about it, we have to use `checked_shl` to disallow overflows. Otherwise, `[0x80; 0x80; 0x80; 0x80; 0x0f]` and `[0x80; 0x80; 0x80; 0x80; 0x7f]` decodes to the same (truncated) u32 value.

Reply:
even better! 👍

Reply:
`checked_shl` does not do what I thought, so I used `u32::try_from`

Reply:
Also, one needs to take care of `[0x80; 0] = [0]`